src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -238,28 +239,84 @@ func (gc *GarbageCollector) mark(ctx job.Context) error {
 	return nil
 }
 
+type SyncError struct {
+	retErr error
+	mu     sync.Mutex
+}
+
+const (
```

Comment:
please leverage the existing util to handle this, https://github.com/goharbor/harbor/blob/cc26d75b56244ee702c49015c120fb660a9d61b5/src/common/utils/passports.go?_pjax=%23repo-content-pjax-container#L77